### PR TITLE
Add precursor mz to selected ion m/z (mzML)

### PR DESCRIPTION
**Pull request by oliveralka**
# Description

1) Throws an error if a MzML without correct SourceFile annotation is used in the SiriusAdapter.

2) Fixes the loss of precursor mz information after conversion of msconvert mzML via FileConverter. 

Fixes: https://github.com/OpenMS/OpenMS/issues/4294

I wanted to discuss the change in the MzMLHandler here first before I proceed with fixing the tests since the change seems to have a wide impact. 

```
The following tests FAILED:
	184 - MzMLFile_test (Failed)
	1219 - TOPP_FeatureFinderMetabo_3_chrom_out (Failed)
	1311 - TOPP_FileConverter_23_out (Failed)
	1321 - TOPP_FileConverter_27_out (Failed)
	1325 - TOPP_FileConverter_29_out (Failed)
	1328 - TOPP_FileConverter_30_out (Failed)
	1330 - TOPP_FileConverter_31_out (Failed)
	1332 - TOPP_FileConverter_32_out (Failed)
	1356 - TOPP_FileFilter_12_out1 (Failed)
	1429 - TOPP_FileFilter_49_out (Failed)
	1695 - TOPP_MapRTTransformer_4_out1 (Failed)
	1782 - TOPP_MRMMapper_test_1_out_1 (Failed)
	1784 - TOPP_MRMMapper_test_2_out_1 (Failed)
	1792 - TOPP_OpenSwathChromatogramExtractor_test_1_out1 (Failed)
	1794 - TOPP_OpenSwathChromatogramExtractor_test_2_out1 (Failed)
	1796 - TOPP_OpenSwathChromatogramExtractor_test_3_out1 (Failed)
	1800 - TOPP_OpenSwathChromatogramExtractor_test_5_out1 (Failed)
	1830 - TOPP_OpenSwathMzMLFileCacher_test_1_out1 (Failed)
	1831 - TOPP_OpenSwathMzMLFileCacher_test_1_out2 (Failed)
	1834 - TOPP_OpenSwathMzMLFileCacher_test_2_out1 (Failed)
	1835 - TOPP_OpenSwathMzMLFileCacher_test_2_out2 (Failed)
	1838 - TOPP_OpenSwathMzMLFileCacher_test_3_out1 (Failed)
	1841 - TOPP_OpenSwathMzMLFileCacher_test_4_out1 (Failed)
	1844 - TOPP_OpenSwathMzMLFileCacher_test_5_out1 (Failed)
	1845 - TOPP_OpenSwathMzMLFileCacher_test_5_out2 (Failed)
	1911 - TOPP_OpenSwathFileSplitter_1_out2 (Failed)
	1928 - TOPP_NoiseFilterGaussian_2_out1 (Failed)
	1932 - TOPP_NoiseFilterGaussian_4_out1 (Failed)
	1936 - TOPP_NoiseFilterSGolay_2_out1 (Failed)
	1940 - TOPP_NoiseFilterSGolay_4_out1 (Failed)
	1952 - TOPP_PeakPickerHiRes_2_out1 (Failed)
	1956 - TOPP_PeakPickerHiRes_4_out1 (Failed)
	1958 - TOPP_PeakPickerHiRes_5_out1 (Failed)
Errors while running CTest
```

### CODE REVIEW

File: src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp [hunk 1]
```diff
@@ -472,9 +472,20 @@ namespace OpenMS
 
     AccessionInfo ainfo;
 
-    // sourcefile 
-    ainfo.sf_path = spectra.getSourceFiles()[0].getPathToFile();
-    ainfo.sf_type = spectra.getSourceFiles()[0].getFileType();
+    // sourcefile
+    if (spectra.getSourceFiles().size() == 0)
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp:476]: .empty() ?

**Comment by oliveralka** [comment]
The "indexListOffset" have changed as well due to a higher filesize - should be ok, or?

**Comment by jpfeuffer** [comment]
Yes, this always happens.

**Comment by oliveralka** [comment]
rebuild jenkins

**Comment by timosachsenberg** [comment]
rebuild jenkins

**Comment by hroest** [comment]
> The "indexListOffset" have changed as well due to a higher filesize - should be ok, or?

yes that is fine and expected to happen

**Comment by timosachsenberg** [comment]
rebuild jenkins

**Pull request merged**